src/schema/rules/sidecars/nirs.yaml

```diff
@@ -0,0 +1,150 @@
+    - json.AnatomicalLandmarkCoordinateSystem == "other"
+  fields:
+    AnatomicalLandmarkCoordinateSystemDescription: required
+
+FiducialsCoordinateSystemDescriptionRec:
+  selectors:
+    - datatype == "nirs"
+    - suffix == "coordsystem"
+    - json.FiducialsCoordinateSystem != "other"
+  fields:
+    FiducialsCoordinateSystemDescription:
+      level: recommended
+      level_addendum: required if FiducialsCoordinateSystem is "other"
+
+FiducialsCoordinateSystemDescriptionReq:
+  selectors:
+    - datatype == "nirs"
+    - suffix == "coordsystem"
+    - json.FiducialsCoordinateSystem == "other"
+  fields:
+    FiducialsCoordinateSystemDescription: required
+
+NirsBase:
+  selectors:
+    - datatype == "nirs"
+    - suffix == "nirs"
+  fields:
+    TaskName: required
+    InstitutionName: recommended
+    InstitutionAddress: recommended
+    Manufacturer: recommended
+    ManufacturersModelName: recommended
+    SoftwareVersions: recommended
+    TaskDescription: recommended
+    Instructions: recommended
+    CogAtlasID: recommended
+    CogPOID: recommended
+    DeviceSerialNumber: recommended
+    RecordingDuration: recommended
+    HeadCircumference: recommended
+    HardwareFilters: recommended
+    SubjectArtefactDescription: recommended
+
+NirsRequired:
+  selectors:
+    - datatype == "nirs"
+    - suffix == "nirs"
+  fields:
+    CapManufacturer:
+      level: recommended
+      description_addendum: |
+        If no cap was used, such as with optodes
+        that are directly taped to the scalp, then the string `none` MUST be used and the `NIRSPlacementScheme` field
+        MAY be used to specify the optode placement.
+    CapManufacturersModelName:
+      level: recommended
+      description_addendum: |
+        If there is no official model number then a description may be provided (for example, `Headband with print
+        (S-M)`). If a cap from a manufacturer was modified, then the field MUST be set to `custom`. If no cap
+        was used, then the `CapManufacturer` field MUST be `none` and this field MUST be `n/a`.")
+    SamplingFrequency__nirs:
+      level: required
+      description_addendum: |
+        Sampling frequency (in Hz) of all the data in the recording, regardless of their type (for example, `12`). If
+        individual channels have different sampling rates, then the field here MUST be specified as `n/a` and the
+        values MUST be specified in the `sampling_frequency` column in channels.tsv.")
+    NIRSChannelCount: required
+    NIRSSourceOptodeCount: required
+    NIRSDetectorOptodeCount: required
+    # Following counts required conditions enforced in checks
+    ACCELChannelCount:
+      level: recommended
+      level_addendum: required if any channel type is ACC
+    GYROChannelCount:
+      level: recommended
+      level_addendum: required if any channel type is GYRO
+    MAGNChannelCount:
+      level: recommended
+      level_addendum: required if any channel type is MAGN
```

Comment:
@rob-luke would it be possible to have all `*ChannelCount` metadata at **maximum** RECOMMENDED? -- But never required?

Personally, I'd even go further and make all ChannelCount metadata OPTIONAL at most.

see also: https://github.com/bids-standard/bids-specification/issues/1277

Reply:
Thanks @sappelhoff, good suggestion.

I find that the `NIRSChannelCount` is one of the most important fields. When discussing datasets and individual measurements, the first question I get from collaborators is usually "how many channels does it have?".  Given that this information is always available in the underlying data file, and the broad usage of this field I think it should be required.

Counting the individual rows in the `channels.tsv` file is not human-friendly, particularly when there are thousands of channels. Further, the number of sources and detectors is not easily read by a human (e.g. it is not always half the number of channels, you can have 60 sources, 100 detectors, and 2000 channels).

I think we should keep the core fNIRS `ChannelCount`fields as REQUIRED, and the auxiliary types, such as GYRO, as OPTIONAL. I understand your concern about mixing requirements, but I think the usefulness of the BEP will be greatly reduced if this information is not stored explicitly in metadata.

Reply:
okay, taken together with the discussion in #1277 I am fine with resolving this here. Thanks!